### ISSUE TITLE
Use dirs or directories crates to get platform-appropriate directories to save output

### ISSUE THREAD

**Issue by kamalmarhubi** [rbspy/rbspy#167]
Once we land #165 we'll have some kind of messy code dealing with generating directories. We should "graduate" to using either [dirs](https://crates.io/crates/dirs) or [directories](https://crates.io/crates/directories). This will clean up the code, and also make rbspy a better citizen. :-)

This will change where files are being saved, as files will move on macOS (to `~/Library/Caches`), and on Linux where a user customises any of the XDG env vars.

**Comment by jeremyjung** [comment]
Hi @kamalmarhubi. I can look into this.  Did you have a preference on which crate to use?

**Comment by acj** [comment]
Hey @jeremyjung, thanks for the offer from ages ago. :)  I've added this to the 0.7 milestone as it would be nice to have. Are you still interested in working on it? PRs are definitely welcome and appreciated.

I don't have a strong opinion on which crate to use. They both look well-maintained. Given our fairly simple needs for file/directory work, I'd probably suggest starting with `directories` to keep a high level of abstraction and only switch to `dirs` if we need to.

**Comment by jeremyjung** [comment]
I can work on it in a week or two. It has been a long time since I’ve looked at rust so I will be... rusty

**Comment by jeremyjung** [comment]
sorry @acj i forgot about this :/ someone else is free to take it!

**Comment by acj** [comment]
All good! Thanks for following up.
